orchagent/nhgorch.h

```diff
@@ -0,0 +1,242 @@
+#pragma once
+
+#include "orch.h"
+#include "nexthopgroupkey.h"
+
+class NextHopGroupMember
+{
+public:
+    /* Constructors / Assignment operators. */
+    NextHopGroupMember(const NextHopKey& nh_key) :
+        m_nh_key(nh_key),
+        m_gm_id(SAI_NULL_OBJECT_ID) {}
+
+    NextHopGroupMember(NextHopGroupMember&& nhgm) :
+        m_nh_key(std::move(nhgm.m_nh_key)),
+        m_gm_id(nhgm.m_gm_id)
+    { nhgm.m_gm_id = SAI_NULL_OBJECT_ID; }
+
+    NextHopGroupMember& operator=(NextHopGroupMember&& nhgm);
+
+    /*
+     * Prevent object copying so we don't end up having multiple objects
+     * referencing the same SAI objects.
+     */
+    NextHopGroupMember(const NextHopGroupMember&) = delete;
+    void operator=(const NextHopGroupMember&) = delete;
+
+    /* Destructor. */
+    virtual ~NextHopGroupMember();
+
+    /* Update member's weight and update the SAI attribute as well. */
+    bool updateWeight(uint32_t weight);
+
+    /* Sync / Desync. */
+    void sync(sai_object_id_t gm_id);
+    void desync();
+
+    /* Getters / Setters. */
+    inline const NextHopKey& getNhKey() const { return m_nh_key; }
+    inline uint32_t getWeight() const { return m_nh_key.weight; }
+    sai_object_id_t getNhId() const;
+    inline sai_object_id_t getGmId() const { return m_gm_id; }
+    inline bool isSynced() const { return m_gm_id != SAI_NULL_OBJECT_ID; }
+
+    /* Check if the next hop is labeled. */
+    inline bool isLabeled() const { return !m_nh_key.label_stack.empty(); }
+
+    /* Convert member's details to string. */
+    std::string to_string() const
```

Comment:
use camelCase, "toString()"

Reply:
Nexthopkey.h and nexthopgroupkey.h use to_string, so think it's cleanest to use the same here (as it's built on top of them)